srcs/incs/parser/Analyser.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef SRCS_INCS_PARSER_ANALYSER_HPP_
+#define SRCS_INCS_PARSER_ANALYSER_HPP_
+
+# include <iostream>
+# include <fstream>
+# include <cstring>
+# include <string>
+# include <sstream>
+
+class Analyser {
+ public:
+	Analyser(void);
+	Analyser(Analyser const &rhs);
+	virtual ~Analyser(void);
+	Analyser	&operator=(Analyser const &rhs);
+	class SyntaxError : public std::exception {
+	public:
+#ifdef __PLATFORM_LINUX
+		virtual ~SyntaxError(void) _GLIBCXX_TXN_SAFE_DYN _GLIBCXX_NOTHROW;
+#elif __PLATFORM_MACOS
+		virtual ~SyntaxError(void) _NOEXCEPT;
+#endif
```

Comment:
Is there any advantage to use `_GLIBCXX_TXN_SAFE_DYN _GLIBCXX_NOTHROW` / `_NOEXCEPT` instead of `throw()`?
```
virtual ~SyntaxError() throw() {}
```
We could replace `__PLATFORM_LINUX` / `__PLATFORM_MACOS` with the standard `__APPLE__` /` __linux__` macros, and this way, we don't have to define them in the Makefile.

Reply:
La verdad es que las usaba porque era lo que mi compilador me decía que la clase de la que heredaba (std::except) tenía. No tenía ni idea de que era, conceptualmente, similar a simplemente throw().
Snippet de fuentes que he encontrado que describe estas macros.
Primero _GLIBCXX_TXN_SAFE_DYN:
```
#if __cplusplus >= 201103L && _GLIBCXX_USE_CXX11_ABI			\
--
&& _GLIBCXX_USE_DUAL_ABI && __cpp_transactional_memory >= 201505L	\
&&  !_GLIBCXX_FULLY_DYNAMIC_STRING && _GLIBCXX_USE_WEAK_REF		\
&& _GLIBCXX_USE_ALLOCATOR_NEW
#define _GLIBCXX_TXN_SAFE transaction_safe
#define _GLIBCXX_TXN_SAFE_DYN transaction_safe_dynamic
#else
#define _GLIBCXX_TXN_SAFE
#define _GLIBCXX_TXN_SAFE_DYN
#endif
```
https://code.woboq.org/qt5/include/c++/8.2.1/x86_64-pc-linux-gnu/bits/c++config.h.html
C++11 ?? No parece que sea útil.
Por otra parte, para _GLIBCXX_NOTHROW
```
#ifndef _GLIBCXX_NOTHROW
--
# ifndef __cplusplus
#  define _GLIBCXX_NOTHROW __attribute__((__nothrow__))
# endif
#endif
```
throw() no parece equivalente a __attribute__((__nothrow__)). Parece lo contrario, de hecho.

Pero si throw() es portable y nos soluciona el problema, está listo.

Reply:
Yo tambien lo encontro un poco confuso. Pense que `throw()` implica que el metodo puede generar una excepcion.
Parece que `throw()` es equivalente a `throw(void)` y, significa que el metodo no genera ninguna excepion.
Y si tiene parametros, por ejemplo `throw(int, char, etc...)`, significa que puede generar excepciones de esos tipos.